core/src/main/scala/org/apache/spark/scheduler/local/LocalBackend.scala

```diff
@@ -74,10 +77,20 @@ private[spark] class LocalActor(
 
   def reviveOffers() {
     val offers = Seq(new WorkerOffer(localExecutorId, localExecutorHostname, freeCores))
-    for (task <- scheduler.resourceOffers(offers).flatten) {
-      freeCores -= scheduler.CPUS_PER_TASK
-      executor.launchTask(executorBackend, taskId = task.taskId, attemptNumber = task.attemptNumber,
-        task.name, task.serializedTask)
+    val tasks = scheduler.resourceOffers(offers).flatten
+    if (tasks.nonEmpty) {
```

Comment:
We won't schedule a reviveOffer() if some tasks will be scheduled, the reviveOffer() will be call once any task finish.